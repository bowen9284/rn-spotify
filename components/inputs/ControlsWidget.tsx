import React, { useContext, useState } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { AuthContext } from '../../context/authContext';

interface Props {
  isFollowed: boolean;
  onFollowPress: (toggle: boolean) => void;
  onDownloadPress: () => void;
  onEllipsisPress: () => void;
}
const ControlsWidget: React.FC<Props> = ({
  isFollowed,
  onFollowPress,
  onDownloadPress,
  onEllipsisPress,
}) => {
  const context = useContext(AuthContext);

  let isFollowedIcon = () => {
    return isFollowed ? (
      <Ionicons name="ios-heart" size={30} color="green" />
    ) : (
      <Ionicons name="heart-outline" size={30} color="gray" />
    );
  };

  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => {
          onFollowPress(!isFollowed);
        }}
      >
        {isFollowedIcon()}
      </TouchableOpacity>
      <TouchableOpacity onPress={() => {}}>
        <Ionicons name="arrow-down-circle-outline" size={30} color="gray" />
      </TouchableOpacity>
      <TouchableOpacity onPress={() => {}}>
        <Ionicons name="ellipsis-horizontal" size={30} color="gray" />
      </TouchableOpacity>
    </View>
  );
};

export default ControlsWidget;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
});
