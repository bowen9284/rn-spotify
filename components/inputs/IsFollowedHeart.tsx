import React from 'react';
import { Ionicons } from '@expo/vector-icons';

type Props = {
  isFollowed: Boolean;
  iconSize?: number;
};

const IsFollowedHeart: React.FC<Props> = ({ isFollowed, iconSize = 30 }) => {
  return isFollowed ? (
    <Ionicons name="ios-heart" size={iconSize} color="#1DB954" />
  ) : (
    <Ionicons name="heart-outline" size={iconSize} color="gray" />
  );
};

export default IsFollowedHeart;
