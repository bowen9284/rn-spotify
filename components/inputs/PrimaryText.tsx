import React from 'react';
import { Text } from 'react-native';
import { useTheme } from '@react-navigation/native';

type Props = {
  children: React.ReactNode;
  style?: {};
};

export const PrimaryText: React.FC<Props> = ({ children, style }) => {
  const { colors } = useTheme();
  return <Text style={{ ...style, color: colors.text }}>{children}</Text>;
};
