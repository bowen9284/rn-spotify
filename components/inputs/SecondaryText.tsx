import React from 'react';
import { Text } from 'react-native';

type Props = {
  children: React.ReactNode;
  style?: {};
  numberOfLines?: number;
};

export const SecondaryText: React.FC<Props> = ({
  children,
  style,
  numberOfLines,
}) => {
  return (
    <Text
      numberOfLines={numberOfLines}
      ellipsizeMode="tail"
      style={{ fontSize: 13, color: 'lightgray', ...style }}
    >
      {children}
    </Text>
  );
};
