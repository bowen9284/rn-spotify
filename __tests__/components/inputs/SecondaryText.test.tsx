import React from "react";
import renderer from "react-test-renderer";
import { PrimaryText } from "../../../components/inputs/PrimaryText";
import { SecondaryText } from "../../../components/inputs/SecondaryText";

const props = {
  children: "some secondary text",
  style: {},
  numberofLines: 2
};

test("renders correctly", () => {
  const tree = renderer.create(<SecondaryText {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
