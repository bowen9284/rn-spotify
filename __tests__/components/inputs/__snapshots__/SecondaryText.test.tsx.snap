// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  ellipsizeMode="tail"
  style={
    Object {
      "color": "lightgray",
      "fontSize": 13,
    }
  }
>
  some secondary text
</Text>
`;
